### ISSUE TITLE
SFTP Get, getting zero byte files

### ISSUE THREAD

**Issue by jpass39** [theophilusx/ssh2-sftp-client#177]
I am trying to use SFTP.get and I continously only get zero bytes file.
```
                            let remotePath = path.join(source, list.name);
                            let localPath = path.join(destination,list.name);
                            let writeStream = fs.createWriteStream(localPath);
                            req.app.locals.io.emit("currentDownload", list.name);
                            try {
                                console.log(source + "/" + list.name);
                                console.log(destination + "/" + list.name);
                                // await sftp.fastGet(source + "/" + list.name, destination + "/" + list.name);
                                await sftp.get(remotePath).then(async (data) => {
                                    let throttle = await stream.createThottle(setting);
                                    return await data.pipe(throttle).pipe(writeStream);
                                });

                            } catch (e) {
                                console.log(e);
                            }
```
not sure if I am doing something wrong, because fastGet if I just pass the file paths works

**Comment by theophilusx** [comment]
It is a little hard to follow what exactly you are attempting to do, but there are a couple of possible issues with your code. 

You don't need to mix promise and async api in the way you have. Probably much clearer to just use the async style to be consistent. In fact, I'm not sure how mixing the two together in the same call would actually work. Try something like (untested - will need debugging)

```
   let remotePath = path.join(source, list.name);
   let localPath = path.join(destination,list.name);
   let writeStream = fs.createWriteStream(localPath);
   req.app.locals.io.emit("currentDownload", list.name);
   try {
     console.log(source + "/" + list.name);
     console.log(destination + "/" + list.name);
    // await sftp.fastGet(source + "/" + list.name, destination + "/" + list.name);
   let data = await sftp.get(remotePath);
   let throttle = await stream.createThottle(setting);
   data.pipe(throttle).pipe(writeStream);
 } catch (e) {
   console.log(e);
 }
```

However, I suspect this will also throw an error as the return value for a single argument call to get() is a Buffer object, not a stream and Buffer objects don't have a pip() method.  I'm not sure what the intention of the stream object is or what createThrottle is all about, but if all you want to do is get the remote data into a writeable stream, possibly doing some sort of processing on the way, then you might find the get-writeable.js example in the exmaple directory of the repository useful. It is using promise style, but easily convertible to async/await style. The concat-streams and through2 npm modules may also be useful, depending on what it is your trying to do. You may also find some of the tests useful as examples - in test/05get.js there is an example of using the zlib module to unzip a gzipped file 'on the fly' as you download it.

**Comment by jpass39** [comment]
I will attempt to change the promise to use async await. I was trying to create a writeable stream and throttle how fast it downloads the remote file by using throttle a npm stream that can limit how fast something downloads.

**Comment by theophilusx** [comment]
I have added a file in the repository called /example/test-get-writeable.js which just downloads a remote file and saves it in your /tmp directory. You call it as 

node ./test-get-writeable.js /path/to/remote/file

See inside the script for a config object you need to edit to set host, username and password. If the script works and the download is successful, then you know the problem is with something in your code. If it fails, then the problem is with something on the server side. This script has been run against some test servers using node 12.11.0 with no errors on both text and binary files.

**Comment by jpass39** [comment]
it was node version causing it I was using 8.9. I am not sure how it would break it but thank you
